mypy/semanal.py

```diff
@@ -564,14 +564,15 @@ def check_function_signature(self, fdef: FuncItem) -> None:
 
     def visit_class_def(self, defn: ClassDef) -> None:
         self.clean_up_bases_and_infer_type_variables(defn)
-        if self.analyze_namedtuple_classdef(defn):
-            return
-        self.setup_class_def_analysis(defn)
+        is_named_tuple = self.analyze_namedtuple_classdef(defn)
+
+        if not is_named_tuple:
+            self.setup_class_def_analysis(defn)
 
-        self.bind_class_type_vars(defn)
+            self.bind_class_type_vars(defn)
 
-        self.analyze_base_classes(defn)
-        self.analyze_metaclass(defn)
+            self.analyze_base_classes(defn)
+            self.analyze_metaclass(defn)
 
         for decorator in defn.decorators:
```

Comment:
I don't (although it would be nice to support namedtuple decorators in the future). I'll push an update to change this once tests pass.

Also, I think it shouldn't be hard to implement https://github.com/python/typing/issues/352 by further adjusting the behavior of this function—I might try that once this PR is approved.